### ISSUE TITLE
Prefer Fewest or Most Followers - Not Working properly - [BUG] Other

### ISSUE THREAD

**Issue by mirokrastanov** [TLDRMissions/TLDRMissions#552]
Hey, 
I've noticed that throughout the past month on Kyrian, Venthyr and Necrolord (since I run a separate profile for them), but it might be the same for Night Fae (I just consider sending just 1 follower to all fae missions good enough, since their troops rock). 
So it's tested only on Kyr, Vent & Necro. Not tested on Fae. 

**Issue** 
**The toggle between "Fewest" or "Most"  followers is not working properly.** 
Selecting "Most" sends out 5 followers, no matter if the missions were doable with 2 + troops... 
Selecting "Fewest" sends out 1 follower, no matter if some would not be doable with 1 and would need more... 
I have not found exceptions to this rule, but I assume that even if there are a few, they are a minority.

**Note** 
I have tested with the "Minimum troops" slider set to all values. Tested with 0, 1, 2, 3, 4 and the results don't change. 
My default for Kyr, Vent & Necro covenants is "Minimum troops" = 2.

**Description**
I believe this is a bug or maybe I am not getting the meaning of "Fewest" / "Most" correctly as from my understanding it needs to test (depending on the "Minimum troops" setting) in the following way:
- for "Fewest" & "Minimum troops" = 2 it needs to test first with: 1follower+2troops, 1follower+3troops, 1follower+4troops, 2followers+2troops, 2followers+3troops, 3followers+2troops, and then finish and if none succeed and return "failed" and then not send it
- for "Most" & "Minimum troops" = 2 is should works the same way, but in reverse: 3f+2tr, 2f+2tr, 2f+3tr, 1f+2tr, 1f+3tr, 1f+4tr and return failed if it can't succeed and not send the mission

**Final Thoughts**
That is all. The addon is working flawlessly, besides this so thanks for that. If I am mistaken and the logic behind "Fewest"/"Most" is working differently please accept my apologies and clarify if possible. 
Thank You! 

**My Current Settings** - Just sent 4 missions with 5 followers in each, with those settings, despite 3 of them easily doable with far less and also despite the "Min troops" set to 2, which should restrict the followers to 3 max..
![image](https://user-images.githubusercontent.com/86717185/185943206-48ce8fe8-8fa9-4a79-b6a2-e5df8f117aeb.png)

**Comment by teelolws** [comment]
Fewest: try with 1 follower + troops, then try 2 followers + troops, then 3, then 4, then 5.

Most: try with 5 followers + troops, then try 4 followers + troops, then 3, then 2, then 1.

Minimum troops = 2 + fewest followers: try 1 follower + 2 troops, then 1 follower + 3 troops, then 1 follower + 4 troops, then 2 followers + 1 troop, then 2 followers + 2 troops...

Minimum troops = 4 + fewest followers: try 1 follower + 4 troops, then 2 followers + 3 troops, then 3 followers + 2 troops...

If the addon has already found a solution for a mission with your followers current level and HP, it will ignore your fewest/most and minimum troops settings and instead use the previous solution it found with your previous settings.

**Comment by mirokrastanov** [comment]
Thank You for the clarification. That last statement explains what is most likely causing this behavior. 

I guess I would reinstall the addon, apply my prefered setting first and do more testing after, to see whether that helps. 
It was just super weird how "fewest" always sent just 1 follower and "most" always - a full party of 5 followers, no matter what the min troops were... But yeah I guess I gotta consider what you mentioned and test everything again. :) 

Anyhow, thanks again. I also looked at some of the closed issues and I saw the idea about the external database and I wish you luck with it and I hope it works out. ;) This addon is great and I'm glad it's improving continuously. Keep it up! 

Cheers!
